e2e/_suites/metricbeat/metricbeat_test.go

```diff
@@ -180,7 +247,30 @@ func InitializeMetricbeatScenarios(ctx *godog.ScenarioContext) {
 //nolint:deadcode,unused
 func InitializeMetricbeatTestSuite(ctx *godog.TestSuiteContext) {
 	ctx.BeforeSuite(func() {
+		setupSuite()
 		log.Trace("Before Metricbeat Suite...")
+
+		var suiteTx *apm.Transaction
+		var suiteParentSpan *apm.Span
+		var suiteContext = context.Background()
+
+		if enableInstrumentation {
+			defer apm.DefaultTracer.Flush(nil)
```

Comment:
I'm flushing right after each major block defining a transaction: a scenario, or this setup initialisation.